File: versioned_docs/version-1.18.x/examples/go/src/data_management_delete_data.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	"time"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get bucket to remove
	bucket, err := client.GetBucket(context.Background(), "bucket-to-remove")
	if err != nil {
		panic(err)
	}

	// Remove only entry with name "example-entry"
	err = bucket.RemoveEntry(context.Background(), "example-entry")
	if err != nil {
		panic(err)
	}

	// Wait for the storage to finish removing blocks
	time.Sleep(1 * time.Second)

	// Remove entire bucket
	err = bucket.Remove(context.Background())
	if err != nil {
		panic(err)
	}
}

File: versioned_docs/version-1.18.x/examples/py/src/data_management_delete_data.py

import asyncio
from reduct import Client, Bucket


async def main():
    # Create a client with the base URL and API token
    async with Client("http://localhost:8383", api_token="my-token") as client:
        # Get bucket to remove
        bucket: Bucket = await client.get_bucket("bucket-to-remove")

        # Delete only entry with name "example-entry"
        await bucket.remove_entry("example-entry")

        # Wait for the storage to finish removing blocks
        await asyncio.sleep(1)

        # Remove entire bucket
        await bucket.remove()


if __name__ == "__main__":
    asyncio.run(main())
